CartPole/cartpole_model.py

```diff
@@ -42,27 +42,30 @@
 The 0-angle state is always defined as pole in upright position. This currently cannot be changed
 """
 
+import yaml
+config = yaml.load(open("config.yml", "r"), Loader=yaml.FullLoader)
+
 # Parameters of the CartPole
 P_GLOBALS = SimpleNamespace()  # "p" like parameters
-P_GLOBALS.m = 0.087  # mass of pole, kg # Checked by Antonio & Tobi
-P_GLOBALS.M = 0.230  # mass of cart, kg # Checked by Antonio
-P_GLOBALS.L = 0.395/2.0  # HALF (!!!) length of pend, m # Checked by Antonio & Tobi
-P_GLOBALS.u_max = 6.21  # max force produced by the motor, N # Checked by Marcin
-P_GLOBALS.M_fric = 6.34  # cart friction on track, N/m/s # Checked by Marcin
-P_GLOBALS.J_fric = 2.5e-4  # friction coefficient on angular velocity in pole joint, Nm/rad/s # Checked by Marcin
-P_GLOBALS.v_max = 0.8  # max DC motor speed, m/s, in absense of friction, used for motor back EMF model # TODO: not implemented in model, but needed for MPC
-
-cart_length = 4.4e-2  # m, checked by Marcin&Asude
-usable_track_length = 44.0e-2  # m, checked by Marcin&Asude
+P_GLOBALS.m = config["cartpole"]["m"]
+P_GLOBALS.M = config["cartpole"]["M"]
+P_GLOBALS.L = float(config["cartpole"]["L"].split("/")[0])/float(config["cartpole"]["L"].split("/")[1])
+P_GLOBALS.u_max = config["cartpole"]["u_max"]
+P_GLOBALS.M_fric = config["cartpole"]["M_fric"]
+P_GLOBALS.J_fric = config["cartpole"]["J_fric"]
+P_GLOBALS.v_max = config["cartpole"]["v_max"]
+
+cart_length = config["cartpole"]["cart_length"]
+usable_track_length = config["cartpole"]["usable_track_length"]
 P_GLOBALS.TrackHalfLength = (usable_track_length-cart_length)/2.0  # m, effective length, by which cart center can move
 
-P_GLOBALS.controlDisturbance = 0.0  # disturbance, as factor of u_max
-P_GLOBALS.controlBias = 0.0  # bias of control input
-P_GLOBALS.sensorNoise = 0.0  # sensor noise added to output of the system TODO: not implemented yet
+P_GLOBALS.controlDisturbance = config["cartpole"]["controlDisturbance"]
+P_GLOBALS.controlBias = config["cartpole"]["controlBias"]
+P_GLOBALS.sensorNoise = config["cartpole"]["sensorNoise"]
+
+P_GLOBALS.g = config["cartpole"]["g"]
+P_GLOBALS.k = float(config["cartpole"]["k"].split("/")[0])/float(config["cartpole"]["k"].split("/")[1])
```

Comment:
This structure is not pretty, but used so that you can type the value as a fraction in config